src/app/code/community/Omise/Gateway/controllers/Callback/ValidateoffsitealipayController.php

```diff
@@ -37,9 +37,11 @@ public function indexAction()
 
         if ($charge->isSuccessful()) {
             $invoice = $order->getInvoice($payment->getLastTransId());
+            $transId = $payment->getLastTransId();
 
-            $order->markAsPaid(
-                $payment->getLastTransId(),
+            // Make sure to avoid marking invoice paid more than once
+            if (!$order->isInvoicePaid($transId)) $order->markAsPaid(
```

Comment:
@jonrandy how about we just add this condition inside the `markAsPaid` method instead?

Reply:
see below